src/generators/jsx/utils/buildBarProps.mjs

```diff
@@ -0,0 +1,51 @@
+import readingTime from 'reading-time';
+import { DOC_API_BLOB_EDIT_BASE_URL } from '../../../constants.mjs';
+import { visit } from 'unist-util-visit';
+
+/**
+ * Builds sidebar navigation for API documentation pages
+ *
+ * @param {Map<string, Array<ApiDocMetadataEntry>>} groupedModules - Modules grouped by API
+ * @param {Array<ApiDocMetadataEntry>} headNodes - Main entry nodes for each API
+ */
+export const buildSideBarDocPages = (groupedModules, headNodes) =>
+  headNodes.map(node => {
+    const moduleEntries = groupedModules.get(node.api);
+
+    return {
+      title: node.heading.data.name,
+      doc: `${node.api}.html`,
+      headings: moduleEntries
+        .filter(entry => entry.heading?.data?.name && entry.heading.depth === 2)
+        .map(entry => [entry.heading.data.name, `#${entry.heading.data.slug}`]),
+    };
+  });
+
+/**
+ * Builds metadata for the sidebar and meta bar
+ *
+ * @param {ApiDocMetadataEntry} head - Main API metadata entry
+ * @param {Array<ApiDocMetadataEntry>} entries - All API metadata entries
+ */
+export function buildMetaBarProps(head, entries) {
```

Comment:
Maybe explain a bit more why we need to append all this text

Reply:
Like what, does `// Extract text content for reading time calculation` not sum it up? `readingTime` expects text, not AST.